processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByQueryEngineV2.java

```diff
@@ -341,6 +349,94 @@ private void initializeGroupingKeyV2Dimension(
     }
   }
 
+  private static class LongGroupByColumnSelectorStrategy implements GroupByColumnSelectorStrategy
+  {
+    @Override
+    public int getGroupingKeySize()
+    {
+      return Longs.BYTES;
+    }
+
+    @Override
+    public void processValueFromGroupingKey(
+        GroupByColumnSelectorPlus selectorPlus, ByteBuffer key, Map<String, Object> resultMap
+    )
+    {
+      final long val = key.getLong(selectorPlus.getKeyBufferPosition());
+      resultMap.put(selectorPlus.getOutputName(), val);
+    }
+
+    @Override
+    public void initColumnValues(ColumnValueSelector selector, int columnIndex, Object[] valuess)
+    {
+      valuess[columnIndex] = ((LongColumnSelector) selector).get();
+    }
+
+    @Override
+    public void initGroupingKeyColumnValue(
+        int keyBufferPosition, int columnIndex, Object rowObj, ByteBuffer keyBuffer, int[] stack
+    )
+    {
+      keyBuffer.putLong(keyBufferPosition, (Long) rowObj);
+      stack[columnIndex] = 1;
+    }
+
+    @Override
+    public boolean checkRowIndexAndAddValueToGroupingKey(
+        int keyBufferPosition, Object rowObj, int rowValIdx, ByteBuffer keyBuffer
+    )
+    {
+      keyBuffer.putLong(keyBufferPosition, (Long) rowObj);
```

Comment:
hm, good catch, I've adjusted this function to just return false for longs/floats, with a comment on why (it only handles row values after idx 0 for multivalued rows, so single-valued numeric implementations just need to return false, the first value is handled by initGroupingKeyColumnValue())